src/components/FileList.tsx

```diff
@@ -0,0 +1,132 @@
+"use client";
+import React from "react";
+import { Box, Stack, Typography, useTheme } from "@mui/material";
+import FileElement from "@/components/FileElement";
+import type { DropzoneState } from '@/types';
+import FileUploaded from '@/classe/File';
+
+/**
+ * Props for the FileList component.
+ *
+ * @interface FileListProps
+ *
+ * @property {FileUploaded[]} uploadedFiles - An array of uploaded files.
+ * @property {(url: string) => void} handleDelete - Function to handle the deletion of a file by its URL.
+ * @property {(show: boolean, image_url: string) => void} handleSetDropzoneState - Function to set the state of the dropzone, including whether to show it and the URL of the image.
+ */
+interface FileListProps {
+  uploadedFiles: FileUploaded[];
+  handleDelete: (url: string) => void;
```

Comment:
This function should probably be internal to this component unless shared with other components.

Reply:
done